### PR TITLE
Implement ptp function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.ptp, which returns the range of values (maximum minus minimum) along a specified axis of the input tensor.

The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/torch/numpy.py [hunk 1]
```diff
@@ -1382,6 +1382,16 @@ def prod(x, axis=None, keepdims=False, dtype=None):
...
 
 
+def ptp(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+    if axis is None:
+        return x.max() - x.min()
+    else:
+        return torch.amax(x, dim=axis, keepdim=keepdims) - torch.amin(
+            x, dim=axis, keepdim=keepdims
+        )
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1394]: I couldn't find ptp native method for torch.

File: keras/src/ops/numpy.py [hunk 2]
```diff
@@ -5456,6 +5456,74 @@ def prod(x, axis=None, keepdims=False, dtype=None):
...
+            axis=self.axis,
+            keepdims=self.keepdims,
+        )
+
+    def compute_output_spec(self, x):
+        dtype = backend.standardize_dtype(x.dtype)
+        return KerasTensor(
+            reduce_shape(x.shape, axis=self.axis, keepdims=self.keepdims),
+            dtype=dtype,
+        )
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:5477]: The boolean case is intentionally omitted because the JAX implementation of ptp does not support boolean inputs.

File: keras/src/backend/numpy/numpy.py [hunk 3]
```diff
@@ -1018,6 +1018,10 @@ def prod(x, axis=None, keepdims=False, dtype=None):
     return np.prod(x, axis=axis, keepdims=keepdims, dtype=dtype)
 
 
+def ptp(x, axis=None, keepdims=False):
+    return np.ptp(x, axis=axis, keepdims=keepdims)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1022]: The boolean case is intentionally omitted because the JAX implementation of ptp does not support boolean inputs.

File: keras/src/ops/numpy_test.py [hunk 4]
```diff
@@ -8681,6 +8718,26 @@ def test_prod(self, dtype):
...
+        expected_dtype = standardize_dtype(jnp.ptp(x_jax).dtype)
+
+        self.assertEqual(
+            standardize_dtype(knp.ptp(x).dtype),
+            expected_dtype,
+        )
+        self.assertEqual(
+            standardize_dtype(knp.Ptp().symbolic_call(x).dtype),
+            expected_dtype,
+        )
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:8739]: The boolean case is intentionally omitted because the JAX implementation of ptp does not support boolean inputs.

File: keras/src/backend/jax/numpy.py [hunk 5]
```diff
@@ -1063,6 +1063,11 @@ def prod(x, axis=None, keepdims=False, dtype=None):
     return jnp.prod(x, axis=axis, keepdims=keepdims, dtype=dtype)
 
 
+def ptp(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+    return jnp.ptp(x, axis=axis, keepdims=keepdims)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/jax/numpy.py:1068]: I intentionally follow JAX semantics here. jnp.ptp does not support boolean inputs

File: keras/src/backend/numpy/numpy.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1022]: I intentionally follow JAX semantics here. jnp.ptp does not support boolean inputs

File: keras/src/ops/numpy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:5477]: I intentionally follow JAX semantics here. jnp.ptp does not support boolean inputs

File: keras/src/ops/numpy_test.py [hunk 6]
```diff
@@ -8681,6 +8718,26 @@ def test_prod(self, dtype):
...
+    @parameterized.named_parameters(named_product(dtype=ALL_DTYPES))
+    def test_ptp(self, dtype):
+        import jax.numpy as jnp
+
+        if dtype == "bool":
+            self.skipTest("ptp doesn't support bool dtype")
+
+        x = knp.ones((1, 1, 1), dtype=dtype)
+        x_jax = jnp.ones((1, 1, 1), dtype=dtype)
+        expected_dtype = standardize_dtype(jnp.ptp(x_jax).dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:8730]: I intentionally follow JAX semantics here. jnp.ptp does not support boolean inputs

File: keras/src/backend/jax/numpy.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/jax/numpy.py:1068]: I intentionally follow JAX semantics here. jnp.ptp does not support boolean inputs
